### ISSUE TITLE
Alternative to Google C2DM dependency

### ISSUE THREAD

**Issue by Doomsdayrs** [mastodon/mastodon-android#5]
This application uses C2DM for cloud messaging.

Can the F-Droid packaging be provided with an in-house service instead?

Image of registration:
![Screenshot_20220413-173702_microG_Services_Core](https://user-images.githubusercontent.com/38189170/163274838-9da9179a-f36f-4a61-bdec-79cd63d65d9f.png)

**Comment by darhma** [comment]
Yes, I agree that it would be useful to have an alternative push notification distributor (I use a device with a custom rom without play services and I didn't get push notifications). I think support for https://unifiedpush.org/ could solve this problem. Personally I tried unifiedpush with two matrix clients (schildichat and fluffychat) and it works very well, while with fedilab (a mastodon client) I encountered some issues.

**Comment by DonPicciotto** [comment]
I agree with darhma, UnifiedPush support for F-Droid build would be great
